Mathlib/Data/Int/Lemmas.lean

```diff
@@ -77,6 +77,22 @@ theorem natAbs_inj_of_nonpos_of_nonneg {a b : ℤ} (ha : a ≤ 0) (hb : 0 ≤ b)
   simpa only [Int.natAbs_neg] using natAbs_inj_of_nonneg_of_nonneg (neg_nonneg_of_nonpos ha) hb
 #align int.nat_abs_inj_of_nonpos_of_nonneg Int.natAbs_inj_of_nonpos_of_nonneg
 
+/-- A specialization of `abs_sub_le_of_nonneg_of_le` for working with the signed subtraction
+  of natural numbers. -/
+theorem natAbs_coe_sub_coe_le_of_le {a b n : ℕ} (a_le_n : a ≤ n) (b_le_n : b ≤ n) :
+    natAbs (b - a : ℤ) ≤ n := by
+  rw [← Nat.cast_le (α := ℤ), coe_natAbs, abs_sub_comm]
+  exact abs_sub_le_of_nonneg_of_le (ofNat_nonneg a) ((Nat.cast_le (α := ℤ)).mpr a_le_n)
+    (ofNat_nonneg b) ((Nat.cast_le (α := ℤ)).mpr b_le_n)
```

Comment:
```suggestion
    natAbs (a - b : ℤ) ≤ n := by
  rw [← Nat.cast_le (α := ℤ), coe_natAbs]
  exact abs_sub_le_of_nonneg_of_le (ofNat_nonneg a) (Nat.cast_le.mpr a_le_n)
    (ofNat_nonneg b) (Nat.cast_le.mpr b_le_n)
```
Why do you switch the order of `a` and `b`?  Also, some of the implicit arguments can be automatically inferred.

Similar comments for the next lemma.

Reply:
`Nat.cast_le` was giving me trouble with not being able to infer the target type before `.mpr` was called on it